Classes/Controller/SessionController.php

```diff
@@ -0,0 +1,156 @@
+<?php
+declare(strict_types=1);
+
+namespace PhpList\RestBundle\Controller;
+
+use Doctrine\Common\Persistence\ObjectRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use PhpList\PhpList4\Core\Bootstrap;
+use PhpList\PhpList4\Domain\Model\Identity\Administrator;
+use PhpList\PhpList4\Domain\Model\Identity\AdministratorToken;
+use PhpList\PhpList4\Domain\Repository\Identity\AdministratorRepository;
+use PhpList\PhpList4\Domain\Repository\Identity\AdministratorTokenRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller provides methods to create and destroy REST API sessions.
+ *
+ * @author Oliver Klee <oliver@phplist.com>
+ */
+class SessionController extends Controller
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager = null;
+
+    /**
+     * @var AdministratorRepository|ObjectRepository
+     */
+    private $administratorRepository = null;
+
+    /**
+     * @var AdministratorTokenRepository|ObjectRepository
+     */
+    private $administratorTokenRepository = null;
+
+    /**
+     * The constructor.
+     */
+    public function __construct()
+    {
+        // This will later be replaced by dependency injection.
+        $this->entityManager = Bootstrap::getInstance()->getEntityManager();
+        $this->administratorRepository = $this->entityManager->getRepository(Administrator::class);
+        $this->administratorTokenRepository = $this->entityManager->getRepository(AdministratorToken::class);
+    }
+
+    /**
+     * Creates a new session (if the provided credentials are valid).
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        $rawRequestContent = $request->getContent();
+        $response = new Response();
+        if (!$this->validateCreateRequest($rawRequestContent, $response)) {
+            return $response;
+        }
+
+        $parsedRequestContent = json_decode($rawRequestContent, true);
```

Comment:
Would it be better to perform the raw request decoding outside of individual controllers in order to centralise decoding handling, and make it easier to manage and extend supported request formats in future?